WordPress/Sniffs/PHP/IniSetSniff.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Sniffs\PHP;
+
+use WordPressCS\WordPress\AbstractFunctionParameterSniff;
+use \PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Checks use of ini_set function with a blacklist for errors.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   x.x.x
+ */
+class IniSetSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * Array of functions that must be checked
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $target_functions = array(
+	 *         (string) Function name.
+	 *     );
+	 */
+	protected $target_functions = array(
+		'ini_set' => array(),
+	);
+
+	/**
+	 * Array of options that are allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $whitelisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'valid_values' = array()
+	 *         )
+	 *     );
+	 */
+	protected $whitelisted_options = array(
+		'auto_detect_line_endings' => array(),
+		'highlight.bg'             => array(),
+		'highlight.comment'        => array(),
+		'highlight.default'        => array(),
+		'highlight.html'           => array(),
+		'highlight.keyword'        => array(),
+		'highlight.string'         => array(),
+		'short_open_tag'           => array(
+			'valid_values' => array( 'true', '1', 'On' ),
+		),
+	);
+
+	/**
+	 * Array of options that are not allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $blacklisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'invalid_values' = array()
+	 *             (string) 'message'
+	 *         )
+	 *     );
+	 */
+	protected $blacklisted_options = array(
+		'max_execution_time' => array(
```

Comment:
Safe for these first two entries, this list is in alphabetic order. Might be an idea to make it completely so ?